pkg/compression/compression.go

```diff
@@ -52,6 +52,19 @@ func AlgorithmByName(name string) (Algorithm, error) {
 	return Algorithm{}, fmt.Errorf("cannot find compressor for %q", name)
 }
 
+// AlgorithmsByNames returns slice of Algorithms from slice of Algorithm Names
+func AlgorithmsByNames(names []string) ([]Algorithm, error) {
```

Comment:
Non-blocking:

At this point I think this can be an internal helper in `copy/single.go`. Adding more lookup functions to `pkg/compression` is a bit unattractive:

Context: Importing `pkg/compression` adds all compression implementations, including `c/storage/pkg/chunked/compression`, as dependencies, and it’s unclear to me that that’s always a good idea. Compare how `transports` contains a registry, and `transports/alltransports` adds the dependencies: the caller can choose which one to use. (OTOH the argument that each consumer of `pkg/compression` should always support all algorithms might be stronger, I don’t have a firm opinion.)

---

An alternative could be to modify `ListEdit.*CompressionAlgorithms` to only contain names — that would also get rid of the “algorithm name is known but we don’t have an implementation” failure paths — but I think we don’t need to tinker with that now. In practice `compressionAlgorithmFromMIMEType`, or the data from `blob/compression.go`, are always supported algorithms, so that error path is not worth worrying about or making larger changes now.